lib/auth/keystore/hsm.go

```diff
@@ -254,6 +252,76 @@ func (c *hsmKeyStore) GetJWTSigner(ca types.CertAuthority) (crypto.Signer, error
 	return nil, trace.NotFound("no local PKCS#11 JWT key pairs found in %s CA for %q", ca.GetType(), ca.GetClusterName())
 }
 
+// NewSSHKeyPair creates and returns a new HSM-backed SSHKeyPair.
+func (c *hsmKeyStore) NewSSHKeyPair() (*types.SSHKeyPair, error) {
+	return newSSHKeyPair(c)
+}
+
+// NewTLSKeyPair creates and returns a new HSM-backed TLSKeyPair.
+func (c *hsmKeyStore) NewTLSKeyPair(clusterName string) (*types.TLSKeyPair, error) {
+	return newTLSKeyPair(c, clusterName)
+}
+
+// NewJWTKeyPair creates and returns a new HSM-backed JWTKeyPair.
+func (c *hsmKeyStore) NewJWTKeyPair() (*types.JWTKeyPair, error) {
+	return newJWTKeyPair(c)
+}
+
+func (c *hsmKeyStore) keySetHasLocalKeys(keySet types.CAKeySet) bool {
```

Comment:
this checks that the `KeySet` claims to have keys in the local HSM
but it does not check that the local HSM actually has those keys

for example, a user could accidentally manually delete the HSM keys (e.g. by resetting it)

Reply:
Related to your other comment, I'm currently intentionally allowing it to think it has local keys and fail later. IMO if the keys have been deleted from under our feet, there should be an obvious error